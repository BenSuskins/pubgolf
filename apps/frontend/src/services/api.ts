import axios from 'axios';
import { clearLocalStorage, getGameIdentifier, getPlayerName, setGameIdentifier, setPlayerName, setPlayerId, getPlayerId} from '@/utils/utils';
import { baseURL } from '@/utils/constants';

const api = axios.create({
  baseURL: baseURL,
});

api.interceptors.request.use(request => {
  console.debug('Starting Request', JSON.stringify(request, null, 2));
  return request;
});

api.interceptors.response.use(response => {
  console.debug('Response:', JSON.stringify(response.data, null, 2));
  return response;
}, error => {
  console.error('Error:', JSON.stringify(error.response?.data || error.message, null, 2));
  return Promise.reject(error);
});

export const createGame = async (name: string) => {
  clearLocalStorage()
  const response = await api.post('/api/v1/games', {
      host: name
    });
  setGameIdentifier(response.data.gameCode);
  setPlayerName(name);
  setPlayerId(response.data.playerId);
  return response.data;
};

export const joinGame = async (identifier: string, name: string) => {
  const response = await api.post(`/api/v1/games/${identifier}/join`, {
    name: name
  });
  setGameIdentifier(identifier);
  setPlayerName(name);
  setPlayerId(response.data.playerId);
  return response.data;
};

export const submitScore = async (hole: number, score: number) => {
  const identifier = getGameIdentifier();
  const playerId = getPlayerId();
  const response = await api.post(`/api/v1/games/${identifier}/players/${playerId}/scores`, {
    hole: hole,
    score: score
  });
  return response.data;
};

export const getPlayers = async () => {
  const identifier = getGameIdentifier();
  const response = await api.get(`/api/v1/games/${identifier}`);
  return response.data;
};
