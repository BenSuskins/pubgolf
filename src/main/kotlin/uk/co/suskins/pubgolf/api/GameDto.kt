package uk.co.suskins.pubgolf.api

data class GameDto(
    val identifier: String,
)

