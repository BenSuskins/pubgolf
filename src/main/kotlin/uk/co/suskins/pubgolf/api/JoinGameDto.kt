package uk.co.suskins.pubgolf.api

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min

data class JoinGameDto(
    @JsonProperty("name")
    @field:Min(2) @field:Max(40)
    val name: String
)
