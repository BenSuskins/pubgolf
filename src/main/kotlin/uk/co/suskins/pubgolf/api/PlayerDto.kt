package uk.co.suskins.pubgolf.api

data class PlayerDto(
    val name: String,
    val scores: List<Int>,
    val totalScore: Int
)
