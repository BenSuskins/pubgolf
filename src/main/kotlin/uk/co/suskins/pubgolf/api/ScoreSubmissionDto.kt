package uk.co.suskins.pubgolf.api

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min

data class ScoreSubmissionDto(
    @field:Min(1) @field:Max(9)
    val hole: Int,
    @field:Min(-10) @field:Max(10)
    val score: Int
)
