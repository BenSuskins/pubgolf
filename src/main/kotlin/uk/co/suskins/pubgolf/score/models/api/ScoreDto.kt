package uk.co.suskins.pubgolf.score.models.api

import jakarta.validation.constraints.*
import java.util.*

data class ScoreDto(
    @NotNull
    val id: UUID,
    @Size(min = 2, max = 20)
    @NotNull
    @NotBlank
    @Pattern(regexp = "^[a-zA-Z]*$")
    val name: String,
    @Min(-10)
    @Max(10)
    val holeOne: Int,
    @Min(-10)
    @Max(10)
    val holeTwo: Int,
    @Min(-10)
    @Max(10)
    val holeThree: Int,
    @Min(-10)
    @Max(10)
    val holeFour: Int,
    @Min(-10)
    @Max(10)
    val holeFive: Int,
    @Min(-10)
    @Max(10)
    val holeSix: Int,
    @Min(-10)
    @Max(10)
    val holeSeven: Int,
    @Min(-10)
    @Max(10)
    val holeEight: Int,
    @Min(-10)
    @Max(10)
    val holeNine: Int,
    @Min(-10)
    @Max(90)
    val total: Int
)
