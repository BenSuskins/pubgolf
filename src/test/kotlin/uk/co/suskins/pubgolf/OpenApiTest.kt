package uk.co.suskins.pubgolf

import org.approvaltests.Approvals
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.web.client.RestTemplate

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
class OpenApiTest {
    @Test
    fun `Can generate an open api spec`() {
        Approvals.verify(openApiDocs())
    }

    private fun openApiDocs(): String? {
        val restTemplate = RestTemplate()
        return restTemplate.getForObject("http://localhost:8080/v3/api-docs", String::class.java)
    }
}