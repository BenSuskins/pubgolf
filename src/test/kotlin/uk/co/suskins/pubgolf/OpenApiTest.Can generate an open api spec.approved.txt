{"openapi":"3.1.0","info":{"title":"Pub Golf","description":"API's to power Pub Golf Application","contact":{"name":"Ben Suskins","email":"devleopment@suskins.co.uk"},"version":"1.0.0"},"servers":[{"url":"http://localhost:8080","description":"Generated server url"}],"paths":{},"components":{}}